### ISSUE TITLE
rewrite the "materialstore" in Java to be independent on the Groovy runtime

### ISSUE THREAD

**Issue by kazurayam** [kazurayam/materialstore#64]
I want use the materialstore library in a Java 8 Selenium project. But it requires dependency to the Groovy runtime. Cumbersome. The materialstore should be pure Java.

**Comment by kazurayam** [comment]
As https://github.com/kazurayam/materialstore/issues/143 wrote,

I have to put `xxxxx.java` code into the `groovy` directory.

Surprise! surprise! surprise!

**Comment by kazurayam** [comment]
I will start migrating the materialstore lib from Groovy to Java.

**Comment by kazurayam** [comment]
How many .groovy file to translate to .java?

```
:~/github/materialstore/src (issue64toJava)
$ tree | grep .groovy | wc
     111     750    7639
:~/github/materialstore/src (issue64toJava)
$ tree | grep .java | wc
      25     186    1601
```

I have 111 .groovy files.
I have 25 .java files.

Still long way to go. If I do 5 classes a day, it will take me 23 days.

**Comment by kazurayam** [comment]
My progress

:~/github/materialstore (issue64toJava)
$ tree | grep .groovy | wc
     106     902    7940
:~/github/materialstore (issue64toJava)
$ tree | grep .java | wc
     112    1096    8358

**Comment by kazurayam** [comment]
current progress

```
:~/github/materialstore (issue64toJava)
$ tree | grep .groovy | wc
      73     582    5462
:~/github/materialstore (issue64toJava)
$ tree | grep .java | wc
     151    1466   11198
```

**Comment by kazurayam** [comment]
current status

```
:~/github/materialstore (issue64toJava)
$ tree | grep .groovy | wc
      32     260    2319
:~/github/materialstore (issue64toJava)
$ tree | grep .java | wc
     193    1796   14336
```

**Comment by kazurayam** [comment]
finished translating *.groovy codes that can be translated into *.java.

published v0.5.0-alpha

Some *.groovy codes that depend on MarkupBuilder remains yet.

**Comment by kazurayam** [comment]
at the v0.5.0-alpha 

```
:~/github/materialstore (develop *)
$ tree | grep .groovy | wc
      35     272    2438
:~/github/materialstore (develop *+)
$ tree | grep .java | wc
     275    2436   20122

```
